lib/agenda.js

```diff
@@ -24,6 +24,7 @@ var Agenda = module.exports = function(config, cb) {
   this._lockedJobs = [];
   this._jobQueue = [];
   this._defaultLockLifetime = config.defaultLockLifetime || 10 * 60 * 1000; // 10 minute default lockLifetime
+  this._sort = config.sort || { priority: -1, _id: 1 };
```

Comment:
Apologies for not being clear enough, I typed that on my phone so I gave up trying to give an example!

[Indexes are automatically created](https://github.com/agenda/agenda/blob/7c4692a1c4586d6078ccd03538c6f6b22896722f/lib/agenda.js#L103) and the order is quite important so Mongo can take advantage of it during sorting. So we should be adding the sort config in the right place, e.g.

```js
this._collection.createIndexes([{
  "key": Object.assign({"name": 1}, config.sort, {"lockedAt": 1, "nextRunAt": 1, "disabled": 1}),
  "name": "findAndLockNextJobIndex1"
} // ...
```